src/main/java/soolutions/hometaskapp/account/BasicAccount.java

```diff
@@ -0,0 +1,34 @@
+package soolutions.hometaskapp.account;
+
+import soolutions.hometaskapp.core.Amount;
+import soolutions.hometaskapp.user.User;
+
+public class BasicAccount implements Account {
+    private Amount amount;
+    private final User user;
+
+    public BasicAccount(User user) {
+      // TODO(vivek): I am not sure if using 'new' is a good idea here
+      this(user, new Amount(0, "USD"));
+    }
+
+    public BasicAccount(User user, Amount amount) {
+      this.user = user;
+      this.amount = amount;
+    }
+
+    @Override
+    public boolean open() {
+        return true;
+    }
+
+    @Override
+    public Amount balance() {
+      return amount;
+    }
+
+    @Override
+    public void deposit(Amount amount) {
+      this.amount = amount.add(this.amount);;
```

Comment:
Is it a good idea to mutate the `amount` field?

Reply:
No,`amount.add` returns a new amount.

Reply:
`this.amount` is mutating an internal value of Account

Reply:
Hmm, so you want me to return a new instance of the `Account`? if yes, can I ask why?

Reply:
That's a very big question to answer. A short summary would be: prevents state bugs, enables concurrency, and is easier to reason about.

Here's a link to other thoughts on the matter: https://stackoverflow.com/questions/214714/mutable-vs-immutable-objects

I'm not saying that it has to be immutable here. Immutability isn't always the right choice. I'm just curious about your thinking for this particular piece of mutable state.

Reply:
This is Yegor's argument for immutable objects: http://www.yegor256.com/2014/06/09/objects-should-be-immutable.html